functions.php

```diff
@@ -501,3 +501,39 @@ function ($canonical_url, $post) {
     10,
     2
 );
+
+add_action(
+    'admin_init',
+    function () {
+        register_setting(
+            'writing',
+            'revisions_to_keep',
+        );
+
+        add_settings_section(
+            'post-revisions-section',
+            '',
+            function () {
+                echo '';
+            },
+            'writing'
+        );
+
+        add_settings_field(
+            'my-settings-field-id',
+            'Post Revisions',
+            function (){
+                echo '
+                    <label for="limit-post-revisions">
+                        <input id="limit-post-revisions" type="number" name="revisions_to_keep" value=' . get_option('revisions_to_keep') . '>  '
```

Comment:
By default `revisionss_to_keep` value is empty at start, so you may pass a default value as a second arg in `get_option()` function.
(https://developer.wordpress.org/reference/functions/get_option/)